devlib/module/thermal.py

```diff
@@ -0,0 +1,108 @@
+# limitations under the License.
+
+import logging
+import re
+
+from devlib.module import Module
+
+class TripPoint(object):
+    def __init__(self, zone, _id):
+        self._id = _id
+        self.zone = zone
+        self.temp_node = 'trip_point_' + _id + '_temp'
+        self.type_node = 'trip_point_' + _id + '_type'
+
+    def get_temperature(self):
+        """Returns the currently configured temperature of the trip point"""
+        temp_file = self.zone.target.path.join(self.zone.path, self.temp_node)
+        return self.zone.target.read_value(temp_file)
+
+    def set_temperature(self, temperature):
+        temp_file = self.zone.target.path.join(self.zone.path, self.temp_node)
+        self.zone.target_write_value(temp_file, temperature)
+
+    def get_type(self):
+        """Returns the type of trip point"""
+        type_file = self.zone.target.path.join(self.zone.path, self.type_node)
+        return self.zone.target.read_value(type_file)
+
+class ThermalZone(object):
+    def __init__(self, target, root, _id):
+        self.target = target
+        self.name = 'thermal_zone' + _id
+        self.path = target.path.join(root, self.name)
+        self.trip_points = {}
+
+        for entry in self.target.list_directory(self.path):
+            re_match = re.match('^trip_point_([0-9]+)_temp', entry)
+            if re_match is not None:
+                self.trip_point(re_match.group(1))
+
+    def trip_point(self, _id):
+        self.trip_points[int(_id)] = TripPoint(self, _id)
+
+    def is_enabled(self):
+        """Returns a boolean representing the 'mode' of the thermal zone"""
+        value = self.target.read_value(self.target.path.join(self.path, 'mode'))
+        if value == 'enabled':
+            return True
+        else:
+            return False
+
+    def set_mode(self, enable):
+        value = 'disable'
+        if enable:
+            value = 'enabled'
+
+        self.target.write_value(self.target.path.join(self.path, 'mode', value))
+
+    def get_temperature(self):
+        """Returns the temperature of the thermal zone"""
+        temp_file = self.target.path.join(self.path, 'temp')
+        return self.target.read_value(temp_file)
+
+class ThermalModule(Module):
+    name = 'thermal'
+    thermal_root = '/sys/class/thermal'
+
+    @staticmethod
+    def probe(target):
+
+        if target.file_exists(ThermalModule.thermal_root):
+            return True
+
+    def __init__(self, target):
+        super(ThermalModule, self).__init__(target)
+
+        self.logger = logging.getLogger('Thermal')
+
+        self.zones = {}
+        self.cdevs = []
+
+        for entry in target.list_directory(self.thermal_root):
+            re_match = re.match('^(thermal_zone|cooling_device)([0-9]+)', entry)
+
+            if re_match.group(1) == 'thermal_zone':
+                self.thermal_zone(re_match.group(2))
+            elif re_match.group(1) == 'cooling_device':
+                pass
```

Comment:
Indeed, it's a TODO. I have implemented what I needed - hopefully over time we can cover most thermal related operations in this module.